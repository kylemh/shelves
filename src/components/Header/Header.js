import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button';
import styles from './Header.module.scss';

export default class Header extends React.Component {
  static propTypes = {
    createShelf: PropTypes.func.isRequired,
  };

  render() {
    const { createShelf } = this.props;

    return (
      <header className={styles.Header}>
        <h1 className={styles.logo}>Shelf</h1>

        <Button onClick={createShelf}>Create Shelf</Button>
      </header>
    );
  }
}
