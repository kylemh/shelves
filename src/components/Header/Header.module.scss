@import '../../styles/variables.scss';

.Header {
  position: static;
  width: 100vw;
  height: 70px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: $feint;
  border-bottom: 3px solid $gray;

  & > * {
    margin: 0 2rem;
  }
}
