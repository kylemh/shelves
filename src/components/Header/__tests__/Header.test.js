import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';

describe('Header', () => {
  it('should render', () => {
    const wrapper = shallow(<Header createShelf={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should fire callback when button is clicked', () => {
    const mockedCallback = jest.fn();
    const wrapper = shallow(<Header createShelf={mockedCallback} />);

    expect(mockedCallback).toBeCalledTimes(0);
    wrapper.find('Button').simulate('click');
    expect(mockedCallback).toBeCalledTimes(1);
  })
});
