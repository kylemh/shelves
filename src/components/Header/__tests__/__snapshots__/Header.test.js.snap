// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<header
  className="Header"
>
  <h1
    className="logo"
  >
    Shelf
  </h1>
  <Button
    disabled={false}
    fullWidth={false}
    onClick={[MockFunction]}
    tabIndex={0}
    theme="primary"
    type="button"
  >
    Create Shelf
  </Button>
</header>
`;
